src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -983,9 +1003,16 @@ okapiHeaders, null, handlePreviousResponse(false, false, false,
           }
 
           if(!aRequestHasFailed[0]){
-            asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
-              PostBlUsersLoginResponse.respond201WithApplicationJson(cu,
-                PostBlUsersLoginResponse.headersFor201().withXOkapiToken(token))));
+            if (isLoginLegacy(loginEndpoint)) {
+              String token = loginResponse[0].get().getHeaders().get(OKAPI_TOKEN_HEADER);
+              asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+                  PostBlUsersLoginResponse.respond201WithApplicationJson(cu,
+                    PostBlUsersLoginResponse.headersFor201().withXOkapiToken(token))));
+            } else {
+              asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+                respond201WithCookieHeadersAndTokenExpiration(loginResponse[0], cu)
+              ));
+            }
```

Comment:
Given that these branches are accessed by two different invocations. Could this be achieved by passing in the desired differences in behaviour (kinda like polymorphism) rather than using this if statement?

Reply:
I had thought about doing something like that but didn't think it was worth the trouble, but your comment gave me the needed encouragement. Instead of the if/else I now have a `BiFunction` as an argument. It makes the method signature a bit more complicated, but it made the two branches more consistent and simplifies forming the succeeded future. See what you think.

Reply:
Ok, thank you for being willing to try this out. 

I'll resolve this comment and follow up if needed on the new code